core/src/main/java/io/kestra/core/runners/pebble/Extension.java

```diff
@@ -29,6 +31,10 @@ public class Extension extends AbstractExtension {
     @Inject
     private ReadFileFunction readFileFunction;
 
+    @Inject
+    @Nullable
```

Comment:
Why is it nullable?

Reply:
Because the function isn't created only if the recursive settings is set to true

Reply:
Oh god!
Maybe it would be better to create it and throw, with this we would be able to display a better error message.
Anyway, OK.

Reply:
Ludo advised not to inject the function and let Pebble say that this function doesn't exist, it will be documented that render function cannot be used with recursive setting